Sources/JExtractSwiftLib/JNI/JNISwift2JavaGenerator+SwiftThunkPrinting.swift

```diff
@@ -320,28 +318,54 @@ extension JNISwift2JavaGenerator {
 
   /// Prints the implementation of the destroy function.
   private func printDestroyFunctionThunk(_ printer: inout CodePrinter, _ type: ImportedNominalType) {
+    let selfPointerParam = JavaParameter(name: "selfPointer", type: .long)
     printCDecl(
       &printer,
       javaMethodName: "$destroy",
       parentName: type.swiftNominal.name,
       parameters: [
-        JavaParameter(name: "selfPointer", type: .long)
+        selfPointerParam
       ],
       isStatic: true,
       resultType: .void
     ) { printer in
+      let parentName = type.qualifiedName
+      let selfVar = self.printSelfJLongToUnsafeMutablePointer(&printer, swiftParentName: parentName, selfPointerParam)
       // Deinitialize the pointer allocated (which will call the VWT destroy method)
       // then deallocate the memory.
       printer.print(
         """
-        let pointer = UnsafeMutablePointer<\(type.qualifiedName)>(bitPattern: Int(Int64(fromJNI: selfPointer, in: environment!)))!
-        pointer.deinitialize(count: 1)
-        pointer.deallocate()
+        \(selfVar).deinitialize(count: 1)
+        \(selfVar).deallocate()
         """
       )
     }
   }
 
+  /// Print the necessary conversion logic to go from a `jlong` to a `UnsafeMutablePointer<Type>`
+  ///
+  /// - Returns: name of the created "self" variable
+  private func printSelfJLongToUnsafeMutablePointer(
+      _ printer: inout CodePrinter,
+      swiftParentName: String, _ selfPointerParam: JavaParameter) -> String {
+    let newSelfParamName = "self$"
+    printer.print(
+      """
+      guard let env$ = environment else {
+        fatalError("Missing JNIEnv in downcall to \\(#function)")
+      }
+      assert(\(selfPointerParam.name) != 0, "\(selfPointerParam.name) memory address was null")
+      let selfBits$ = Int(Int64(fromJNI: \(selfPointerParam.name), in: env$))
+      assert(selfBits$ != 0, "$self memory address was null: \(selfPointerParam.name) = \\(\(selfPointerParam.name))" )
+      guard let \(newSelfParamName) = UnsafeMutablePointer<\(swiftParentName)>(bitPattern: selfBits$) else {
+        fatalError("Missing self pointer in call to \\(#function)!")
```

Comment:
I think we can move the above assert message to this fatalError instead. The `UnsafeMutablePointer` [initializer](https://developer.apple.com/documentation/swift/unsafemutablepointer/init(bitpattern:)-84bz2?changes=_8) will fail if the bitPattern is zero.

Reply:
Good point, done